test/Integration.t.sol

```diff
@@ -0,0 +1,256 @@
+            owner:        PAUSE_PROXY,
+            rewardsToken: address(l1Token),
+            remoteToken:  address(l2Token),
+            l2Proxy:      address(l2Proxy),
+            l1Bridge:     l1Bridge
+        }));
+
+        VestedRewardsDistributionDeployParams memory distributionParams = VestedRewardsDistributionDeployParams({
+            deployer:  address(this),
+            owner:     PAUSE_PROXY,
+            vest:      address(vest),
+            rewards:   address(l1Proxy)
+        });
+        vestedRewardsDistribution = VestedRewardsDistribution(VestedRewardsDistributionDeploy.deploy(distributionParams));
+
+        ProxiesConfig memory cfg = ProxiesConfig({
+            vest:                      address(vest),
+            vestTot:                   100 * 1e18,
+            vestBgn:                   block.timestamp,
+            vestTau:                   100 days,
+            vestedRewardsDistribution: address(vestedRewardsDistribution),
+            l1RewardsToken:            address(l1Token),
+            l2RewardsToken:            address(l2Token),
+            stakingToken:              stakingToken,
+            l1Bridge:                  l1Bridge,
+            minGasLimit:               1_000_000,
+            rewardThreshold:           1 ether,
+            farm:                      address(farm),
+            rewardsDuration:           1 days,
+            initMinGasLimit:           1_000_000,
+            proxyChainlogKey:          "FARM_PROXY_TKA_TKB_BASE",
+            distrChainlogKey:          "REWARDS_DIST_TKA_TKB_BASE"
+        });
+
+        vm.startPrank(PAUSE_PROXY);
+        FarmProxyInit.initProxies(dss, l1GovRelay, address(l1Proxy), address(l2Proxy), l2Spell, cfg);
+        vm.stopPrank();
+
+        // test L1 side of initProxies
+        vestId = vestedRewardsDistribution.vestId();
+        assertEq(vest.usr(vestId),                                            cfg.vestedRewardsDistribution);
+        assertEq(vest.bgn(vestId),                                            cfg.vestBgn);
+        assertEq(vest.clf(vestId),                                            cfg.vestBgn);
+        assertEq(vest.fin(vestId),                                            cfg.vestBgn + cfg.vestTau);
+        assertEq(vest.tot(vestId),                                            cfg.vestTot);
+        assertEq(vest.mgr(vestId),                                            address(0));
+        assertEq(vest.res(vestId),                                            1);
+        assertEq(l1Proxy.minGasLimit(),                                       cfg.minGasLimit);
+        assertEq(l1Proxy.rewardThreshold(),                                   cfg.rewardThreshold);
+        assertEq(dss.chainlog.getAddress("FARM_PROXY_TKA_TKB_BASE"),          address(l1Proxy));
+        assertEq(dss.chainlog.getAddress("REWARDS_DIST_TKA_TKB_BASE"),        cfg.vestedRewardsDistribution);
+
+        l2Domain.relayFromHost(true);
```

Comment:
Just consider this `relayFromHost` would also be executing the L2 side of the spell for the Bridge set up. Nothing really wrong with that but doesn't get isolated in the `setupGateways` function as was probably the intention.

Reply:
This code is the same as in `arbitrum-farms`.
For some reason when adding `l2Domain.relayFromHost(false);` at the end of `setupGateways` it the test fails with a message of "revert: CrossDomainMessenger: message has already been relayed".
I think @telome saw something similar, but not sure.

Reply:
Ok yeah let's leave this open for now. It doesn't affect the audit as not in the scope anyway.